src/main/java/org/springframework/data/hazelcast/repository/query/HazelcastSortAccessor.java

```diff
@@ -69,13 +68,16 @@
                 throw new UnsupportedOperationException("Null handling not implemented: " + order);
             }
 
-            HazelcastPropertyComparator hazelcastPropertyComparator = new HazelcastPropertyComparator(order.getProperty(),
-                    order.isAscending());
-
-            compoundComparator.addComparator(hazelcastPropertyComparator);
+            if (hazelcastPropertyComparator == null) {
+                hazelcastPropertyComparator = new HazelcastPropertyComparator(order.getProperty(),
+                        order.isAscending());
+            } else {
+                hazelcastPropertyComparator.thenComparing(new HazelcastPropertyComparator(order.getProperty(),
```

Comment:
Could you prepare a test case that covers this?

The `thenComparing` method doesn't modify the state of comparator but returns a new one instead making this line a dead code

Reply:
in the first iteration of the loop, hazelcastPropertyComparator is initialized, for the subsequent iterations, thenComparing applies remaining comparators.

Reply:
The point is you never assign a new `Comparator`

```
hazelcastPropertyComparator.thenComparing(new HazelcastPropertyComparator(order.getProperty(), order.isAscending())); // returns a new Comparator which gets immediately GCed
```

I believe this should be instead:
```
hazelcastPropertyComparator = hazelcastPropertyComparator.thenComparing(new HazelcastPropertyComparator(order.getProperty(), order.isAscending()));
```

But regardless, can you prove that it works by creating a test?

Reply:
I tested it locally, yes, the changes which you have suggested is the one, which is correct.
Currently, I am trying to create a test for HazelCastSortAccessor.

```
package org.springframework.data.hazelcast.repository.query;

import com.hazelcast.query.Predicate;
import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.springframework.data.domain.Sort;
import org.springframework.data.keyvalue.core.query.KeyValueQuery;
import org.springframework.util.ObjectUtils;

import java.io.Serializable;
import java.util.Arrays;

public class HazelcastSortAccessorTest {

    private HazelcastSortAccessor hazelcastSortAccessor = null;
    KeyValueQuery<Predicate<String, Foo>> query = new KeyValueQuery<>();

    @Before
    public void setUp() throws Exception {
        hazelcastSortAccessor = new HazelcastSortAccessor();
        query.setSort(new Sort(Sort.Direction.DESC, Arrays.asList("foo","bar")));
    }

    @After
    public void tearDown() throws Exception {
        hazelcastSortAccessor = null;
    }

    @Test
    public void resolve() {
        Assert.assertEquals(hazelcastSortAccessor.resolve(null),null);
        Object object = hazelcastSortAccessor.resolve(query); // this returns a Comparator#lambda
    }

    static class Foo
            implements Serializable {

        String foo;
        String bar;

        public Foo(String foo, String bar) {
            this.foo = foo;
            this.bar = bar;
        }

        @Override
        public int hashCode() {
            return ObjectUtils.nullSafeHashCode(new Object[] {this.foo,this.bar});
        }

        @Override
        public boolean equals(Object obj) {
            if (this == obj) {
                return true;
            }
            if (obj == null) {
                return false;
            }
            if (!(obj instanceof HazelcastSortAccessorTest.Foo)) {
                return false;
            }
            HazelcastSortAccessorTest.Foo other = (HazelcastSortAccessorTest.Foo) obj;
            return ObjectUtils.nullSafeEquals(this.foo, other.foo) && ObjectUtils.nullSafeEquals(this.bar,other.bar);
        }

    }
}
```

![image](https://user-images.githubusercontent.com/18457700/76214522-e9e1d600-6232-11ea-8377-262548a0dcdc.png)

I am trying to write assert statements to inspect the Object ( line no. 34 ), to retrieve the 'sort by foo & then sort by bar'. 

1. Need to commit changes as suggested by you ( actual working version )
2. Need to add assert statements, which will assert the sort order in resolve.